src/main/java/core/fruitshop/services/impl/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package core.fruitshop.services.impl;
+
+import core.fruitshop.db.Storage;
+import core.fruitshop.model.FruitTransaction;
+import core.fruitshop.services.DataExtractor;
+import core.fruitshop.services.FileWorker;
+import core.fruitshop.services.FruitShopService;
+import core.fruitshop.strategy.interfaces.OperationStrategy;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private static final String COLUMN_SEPARATOR = ",";
+    private static final String REPORT_HEADER = "fruit,quantity";
+    private static final String DATA_FILE_HEADER = "type,fruit,quantity";
+
+    private final FileWorker fileWorker;
+    private final DataExtractor dataExtractor;
+    private final OperationStrategy operationStrategy;
+
+    public FruitShopServiceImpl(FileWorker fileWorker, DataExtractor dataExtractor,
+                                OperationStrategy operationStrategy) {
+        this.fileWorker = fileWorker;
+        this.dataExtractor = dataExtractor;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void processData(String fileName) {
+        for (String line : fileWorker.readFromFile(fileName)) {
```

Comment:
you can process the result of the reading from file, this method should have one responsibility

Reply:
I mean, you're passing the file name , not the data you have read from the file, but the method is called processData, so it should actually process data, not work with file